doc/asic_thermal_monitoring_hld.md

```diff
@@ -0,0 +1,255 @@
+# ASIC thermal monitoring High Level Design
+### Rev 0.1
+## Table of Contents
+
+## 1. Revision 
+Rev | Rev	Date	| Author	| Change Description
+---------|--------------|-----------|-------------------
+|v0.1 |01/10/2019  |Padmanabhan Narayanan | Initial version
+
+## 2. Scope
+ASICs typically have multiple internal thermal sensors. This document describes the high level design of a poller for ASIC thermal sensors. It details how the poller may be configured and the export of thermal values from SAI to the state DB.
+
+## 3. Definitions/Abbreviations
+
+Definitions/Abbreviation|Description
+------------------------|-----------
+ASIC|Application Specific Integrated Circuit. In SONiC context ASIC refers to the NPU/MAC.
+PCIe|Peripheral Component Interconnect express
+SAI| Switch Abstraction Interface
+SDK| Software Development Kit
+NOS| Network Operating System
+
+## 4. Overview
+
+Networking switch platforms are populated with a number of thermal sensors which include exteral (i.e. onboard) as well as internal (those located within the component, e.g. CPU, ASIC, DIMM, Transceiver etc..) sensors. Readings from both the external as well as the internal sensors are essential inputs to the thermal/fan control algorithm so as to maintain optimal cooling. While drivers exist to retrive sensor values from onboard and other internal sensors, the ASIC based sensor values are currently retrieved thru the ASIC's SDK. SAI provides the following attributes to retrive the ASIC internal sensors:
+
+|Attribute|Description|
+|---|------|
+|SAI_SWITCH_ATTR_MAX_NUMBER_OF_TEMP_SENSORS| Maximum number of temperature sensors available in the ASIC |
+|SAI_SWITCH_ATTR_TEMP_LIST|List of temperature readings from all sensors|
+|SAI_SWITCH_ATTR_AVERAGE_TEMP|The average of temperature readings over all sensors in the switch|
+|SAI_SWITCH_ATTR_MAX_TEMP|The current value of the maximum temperature retrieved from the switch sensors|
+
+A configurable ASIC sensors poller is introduced that periodically retrieves the ASIC internal sensor values via SAI APIs and populates the state DB. These values may be used by the thermal control functions (via the ThermalBase() platform APIs), SNMP/CLI or other Telemetry purposes.

```

Comment:
How to avoid duplicated entries if on some platform ASIC thermals can be provided both from platform API and SAI?

Reply:
I don't think platform API and SAI are necessarily exclusive (to generate duplicate entries): if there is no SAI support, the poller would be disabled and the platform API is going to query the ASIC's SDK to populate the values (as part of "SONiC Thermal Control Design"). If there is SAI support, the poller would be the preferred way.